cmd/snap/complete.go

```diff
@@ -98,3 +99,62 @@ func (s keyName) Complete(match string) []flags.Completion {
 	})
 	return res
 }
+
+type connectPlugSpec struct {
+	SnapAndName
+}
+
+func (cps connectPlugSpec) Complete(match string) []flags.Completion {
+	// Parse what the user typed so far, it can be either
+	// nothing (""), a "snap", a "snap:" or a "snap:name".
+	parts := strings.SplitN(match, ":", 2)
+
+	// Ask snapd about available interfaces.
+	cli := Client()
+	ifaces, err := cli.Interfaces()
+	if err != nil {
+		return nil
+	}
+
+	snaps := make(map[string]bool)
+
+	var ret []flags.Completion
+
+	var snapPrefix, plugPrefix string
+	switch len(parts) {
+	case 2:
+		// The user typed the colon, means they know the snap they
+		// want; go with that.
+		plugPrefix = parts[1]
+		snaps[parts[0]] = true
+	case 1:
+		// The user started typing a snap name but didn't reach the
+		// colon yet. Offer plugs for snaps with names that start like
+		// that.
+		snapPrefix = parts[0]
+		fallthrough
+	case 0:
+		// note "" prefixes any snap name :-)
+		for _, plug := range ifaces.Plugs {
+			if strings.HasPrefix(plug.Snap, snapPrefix) {
```

Comment:
You are totally right. As discussed on telegram I was confused by the output of `%v`. The logic is simplified now.